### PR TITLE
<LiterallyCanvas /> React component

### PR DESCRIPTION
**Pull request by irskep**
Based on #358 (thanks @idolize!)

* Introduces a React component that can be used 2 ways:
  1. `<LiterallyCanvas imageURLPrefix="blah" someOtherOption="blah" />`
  2. `<LiterallyCanvas lc={lc} />` (pre-initialized `LiterallyCanvas` object)
* Allows late binding of `lc` to a DOM element. This allows you to initialize it before passing it to a React element.
  * new API: `lc.bindToElement(el)`
  * `repaintLayer` is a no-op if no element is bound yet
* Fixes some gulpfile problems introduced by the commonjs patch
* `ReactDOM` falls back to `React` since usage of `React.render` only generates a warning right now and people might not have upgraded yet
* React component demo at `demo/react-component`

# Questions

~~Should I rename `LC.LiterallyCanvas` to `LC.LiterallyCanvasReact`? I'm afraid the existing name will be confusing; [the docs refer to a "LiterallyCanvas" object all over the place.](http://literallycanvas.com/api/LiterallyCanvas.html)~~ I decided to export it as `LiterallyCanvasReactComponent` to be completely unambiguous.

Are there any negative consequences for non-React users now that `lc` may not be bound to an element for one or more frames?

### CODE REVIEW

**Comment by irskep** [comment]
This appears to break canvas rendering somehow, will fix...

**Comment by irskep** [comment]
It wasn't this branch. Never mind, things are good.

File: demo/react-component/react-dom-0.14.6.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
+/**
```
Comment (idolize) [inline demo/react-component/react-dom-0.14.6.js]: Why not include this via `npm`?

**Comment by idolize** [comment]
Awesome! Love it :+1: 

Thanks for getting back into it to make these changes with me. Looking forward to using this once you publish to npm.

File: demo/react-component/react-dom-0.14.6.js (hunk 1 above)
Comment (irskep) [inline demo/react-component/react-dom-0.14.6.js]: The demo dependencies are all checked in so we can test with a variety of React versions independent of what is specified in `package.json`.

**Pull request merged**